src/lsstseries/ensemble.py

```diff
@@ -453,17 +540,35 @@ def from_parquet(
             self._err_col = err_col
         if band_col is not None:
             self._band_col = band_col
+        if provenance_col is not None:
+            self._provenance_col = provenance_col
 
         if additional_cols:
             columns = None  # None will prompt read_parquet to read in all cols
         else:
-            columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
+            if self._provenance_col is not None:
+                columns = [
```

Comment:
This if-else approach is fine for the current set of parameters, but if you are going to include more optional columns then a better approach would be:

```
columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
if self._provenance_col is not None:
    columns.append(self._provenance_col)
```